File: csharp/Behavioral/13-ChainOfResponsibility/Service/JobApprovalService.cs
public class JobApprovalService
{
    private readonly IApprovalHandler _chain;

    public JobApprovalService()
    {
        var handyman = new HandymanApprover();
        var supervisor = new SupervisorApprover();
        var manager = new ManagerApprover();
        var director = new DirectorApprover();

        handyman
            .SetNext(supervisor)
            .SetNext(manager)
            .SetNext(director);

        _chain = handyman;
    }

    public void RequestApproval(ApprovalRequest request)
    {
        Console.WriteLine($"\n--- Approval request: {request} ---");
        _chain.Handle(request);
    }
}

File: typescript/Behavioral/13-ChainOfResponsibility/Service/JobApprovalService.ts
import { IApprovalHandler } from "../Handler/IApprovalHandler";
import { HandymanApprover } from "../Handler/HandymanApprover";
import { SupervisorApprover } from "../Handler/SupervisorApprover";
import { ManagerApprover } from "../Handler/ManagerApprover";
import { DirectorApprover } from "../Handler/DirectorApprover";
import { ApprovalRequest } from "../Model/ApprovalRequest";

export class JobApprovalService {
  private readonly chain: IApprovalHandler;

  constructor() {
    const handyman = new HandymanApprover();
    const supervisor = new SupervisorApprover();
    const manager = new ManagerApprover();
    const director = new DirectorApprover();

    handyman.setNext(supervisor).setNext(manager).setNext(director);

    this.chain = handyman;
  }

  requestApproval(request: ApprovalRequest): void {
    console.log(`\n--- Approval request: ${request} ---`);
    this.chain.handle(request);
  }
}
